smqtk_descriptors/impls/descriptor_generator/prepopulated.py

```diff
@@ -0,0 +1,124 @@
+# copy of pytorch
+import abc
+import copy
+import itertools
+import logging
+from typing import (
+    Any, Callable, Dict, Iterable, Iterator, Literal, Mapping,
+    Optional, Sequence, Set, Type, TypeVar, Union
+)
+
+import numpy as np
+
+from smqtk_dataprovider import DataElement
+from smqtk_descriptors import DescriptorGenerator
+from smqtk_image_io import ImageReader
+from smqtk_core.configuration import (
+    from_config_dict,
+    make_default_config,
+    to_config_dict,
+)
+from smqtk_descriptors.utils.parallel import parallel_map
+
+LOG = logging.getLogger(__name__)
+
+__all__ = [
+    "PrePopulatedDescriptorGenerator"
+
+]
+T = TypeVar("T", bound="PrePopulatedDescriptorGenerator")
+
+
+
+
+class PrePopulatedDescriptorGenerator (DescriptorGenerator):
+
+
+    @classmethod
+    def is_usable(cls) -> bool:
+        return True
+
+    @classmethod
+    def get_default_config(cls) -> Dict[str, Any]:
+        c = super().get_default_config()
+        return c
+
+    @classmethod
+    def from_config(
+        cls: Type[T],
+        config_dict: Dict,
+        merge_default: bool = True
+    ) -> T:
+        # Copy config to prevent input modification
+        config_dict = copy.deepcopy(config_dict)
+        return super().from_config(config_dict, merge_default)
+
+    def __init__(
+        self,
+    ):
+        super().__init__()
+
+    @abc.abstractmethod # might be internal and never called
+    def _load_module(self) -> "torch.nn.Module":
+        """
+        :return: Load and return the module.
+        """
+
+    def __getstate__(self) -> Dict[str, Any]:
+        return self.get_config()
+
+    def __setstate__(self, state: Mapping[str, Any]) -> None:
+        # This ``__dict__.update`` works because configuration parameters
+        # exactly match up with instance attributes currently.
+        self.__dict__.update(state)
+
+
+
+    def valid_content_types(self) -> Set:
+        return set()
+
+    def is_valid_element(self, data_element: DataElement) -> bool:
+        # Check element validity though the ImageReader algorithm instance
+        return False
+
+
+    def _generate_arrays(self, data_iter: Iterable[DataElement]) -> Iterable[np.ndarray]:
+        # Generically load image data [in parallel], iterating results into
+        # template method.
+        ir_load: Callable[..., Any] = self.image_reader.load_as_matrix
+        i_load_threads = self.image_load_threads
+
+        gen_fn = (
+            self.generate_arrays_from_images_naive,     # False
+            self.generate_arrays_from_images_iter,      # True
+        )[self.iter_runtime]
+
+        if i_load_threads is None or i_load_threads > 1:
+            return gen_fn(
+                parallel_map(ir_load, data_iter,
+                             cores=i_load_threads)
+            )
+        else:
+            return gen_fn(
+                ir_load(d) for d in data_iter
+            )
+
```

Comment:
```suggestion
```

Reply:
Inheriting from `DescriptorGenerator` has the abstract method `_generate_arrays()` which will require an implementation.  The implementation method will raise an exception error stating that the method should not be called since descriptors are prepopulated.